apps/api/src/app/workflows-v2/usecases/upsert-workflow/upsert-workflow.usecase.ts

```diff
@@ -168,6 +178,78 @@ export class UpsertWorkflowUseCase {
     );
   }
 
+  private async validateWorkflowUniqueness(
+    existingWorkflow: NotificationTemplateEntity | null,
+    command: UpsertWorkflowCommand
+  ): Promise<void> {
+    const { name } = command.workflowDto;
+    const { environmentId, organizationId } = command.user;
+
+    await this.validateWorkflowIdUniqueness(existingWorkflow, command, name);
+
+    await this.validateWorkflowNameUniqueness(name, environmentId, organizationId, existingWorkflow, command);
```

Comment:
George, according to this https://linear.app/novu/issue/NV-4521/treat-workflowid-and-stepid-as-read-only-values we need to treat only workflow Id and step ID as unique. 

Why did we add name uniqueness?

Reply:
Good point! I think we should clarify this in the ticket. In our [discussion](https://novu.slack.com/archives/C07HVLG7G2U/p1729510091455469?thread_ts=1729509114.021689&cid=C07HVLG7G2U), we mentioned that the name should be unique as well to improve the user experience in the future. We could help the user by automatically adding a suffix to the name in such cases. For example, if "Welcome Notification" is created twice, the second one would be named "Welcome Notification (1)." or something similiar